src/storage/Serializer.h

```diff
@@ -0,0 +1,28 @@
+// See the file "COPYING" in the main distribution directory for copyright.
+
+#pragma once
+
+#include "zeek/Span.h"
+#include "zeek/Val.h"
+
+namespace zeek::storage {
+
+namespace detail {
+using byte_buffer = std::vector<std::byte>;
+using byte_buffer_span = Span<const std::byte>;
+} // namespace detail
+
+class Serializer {
+public:
+    virtual ~Serializer() = default;
+    virtual std::optional<detail::byte_buffer> Serialize(ValPtr val) = 0;
```

Comment:
Over in `cluster`, the Serialize() method takes a `detail::byte_buffer&` that can be pre-allocated and re-used. With this API, the implementer definitely needs to allocate a new `detail::byte_buffer` and cannot re-use a pre-allocated one. Nothing to do, but wanted to point out why it's different elsewhere.